go-controller/pkg/ovn/controller/services/services_controller.go

```diff
@@ -207,6 +208,47 @@ func (c *Controller) Run(workers int, stopCh <-chan struct{}, runRepair, useLBGr
 		c.repair.runBeforeSync(c.useTemplates)
 	}
 
+	// Ensure service node cache is up-to-date before we allow service sync to start.
+	// This avoids a case where ovnkube-controller may be restarting, the node cache is not up-to-date,
+	// a service is rendered with a node missing, and its per node LB is incorrectly removed.
+	maxTimeout := 10 * time.Second
+	klog.Infof("Waiting up to %s for service node cache to be synced", maxTimeout)
+	start := time.Now()
+	var err1 error
+	err = wait.PollUntilContextTimeout(ctx, 250*time.Millisecond, maxTimeout, true, func(ctx context.Context) (bool, error) {
+		nodes, err := c.nodeInformer.Lister().List(labels.Everything())
+		if err != nil {
+			err1 = fmt.Errorf("failed to get nodes from lister: %v", err)
+			return false, nil
+		}
+		c.nodeInfoRWLock.Lock()
```

Comment:
apparently in hypershift we can have 0 nodes in the cluster so if len(nodes) == 0 return from here and no need to hold unhold cache etc..